src/snowflake/cli/plugins/snowpark/package/commands.py

```diff
@@ -131,46 +133,127 @@ def package_upload(
+)
+
+index_option = typer.Option(
+    None,
+    "--index-url",
+    help="Base URL of the Python Package Index to use for package lookup. This should point to "
+    " a repository compliant with PEP 503 (the simple repository API) or a local directory laid"
+    " out in the same format.",
+    show_default=False,
+)
+
+skip_version_check_option = typer.Option(
+    False,
+    "--skip-version-check",
+    help="Skip comparing versions of dependencies between requirements and Anaconda.",
+)
+
+allow_shared_libraries_option = typer.Option(
+    False,
+    "--allow-shared-libraries",
+    help="Allows shared (.so) libraries, when using packages installed through PIP.",
 )
 
 
 @app.command("create", requires_connection=True)
-@cleanup_after_install
+@cleanup_packages_dir
 def package_create(
     name: str = typer.Argument(
         ...,
         help="Name of the package to create.",
     ),
-    install_packages: bool = install_option,
+    ignore_anaconda: bool = ignore_anaconda_option,
+    index_url: Optional[str] = index_option,
+    skip_version_check: bool = skip_version_check_option,
+    allow_shared_libraries: bool = allow_shared_libraries_option,
+    deprecated_allow_native_libraries: PypiOption = deprecated_allow_native_libraries_option,
     _deprecated_install_option: bool = deprecated_install_option,
-    allow_native_libraries: PypiOption = PackageNativeLibrariesOption,
+    _deprecated_install_packages: bool = deprecated_pypi_download_option,
     **options,
 ) -> CommandResult:
     """
     Creates a Python package as a zip file that can be uploaded to a stage and imported for a Snowpark Python app.
     """
-    if _deprecated_install_option:
-        install_packages = _deprecated_install_option
+    allow_shared_libraries_pypi_option = {
+        True: PypiOption.YES,
+        False: PypiOption.NO,
+    }[allow_shared_libraries]
+    if deprecated_allow_native_libraries != PypiOption.NO:
+        allow_shared_libraries_pypi_option = deprecated_allow_native_libraries
+
+    if ignore_anaconda:
+        anaconda = None
+    else:
+        anaconda = AnacondaChannel.from_snowflake()
+        package = Requirement.parse(name)
+        if anaconda.is_package_available(
+            package, skip_version_check=skip_version_check
+        ):
+            return MessageResult(
+                f"Package {name} is already available in Snowflake Anaconda Channel."
+            )
 
-    lookup_result = lookup(
-        name=name,
-        install_packages=install_packages,
-        allow_native_libraries=allow_native_libraries,
+    packages_dir = SecurePath(".packages")
+    packages_are_downloaded, dependencies = download_packages(
+        anaconda=anaconda,
+        perform_anaconda_check_for_dependencies=not ignore_anaconda,
+        package_name=name,
+        requirements_file=None,
+        packages_dir=packages_dir,
+        index_url=index_url,
+        allow_shared_libraries=allow_shared_libraries_pypi_option,
```

Comment:
I'm wondering if we have to push this deep into logic, how about structuing this command as
```
download_packages()
native = check_for_native_deps()
if native deps:
    do_what_neccessary()
  
  create_zip_package()
  ```
  
  WDYT? I think pushing all those parameters deep down and mixing flow and action control cause a lot of confusion

Reply:
I wouldn't move the logic outside `download_packages` as is is reused in `snowpark build` command. I changed `if` statements a little:
```
if detected_native_libraries():
    if not _confirm_action()
        FAIL
extract_necessary_packages()
return OK